### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by TheBigLee**
## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog

### CODE REVIEW

File: dump-objects [hunk 1]
```diff
@@ -20,6 +20,8 @@ readonly min_expected_kinds=98
 
 default_output_dir=/data/k8s-backup
 
+DEBUG=false
```
Comment (mhutter) [inline dump-objects]: This cannot be overwritten from outside

**Review by mhutter — CHANGES_REQUESTED** [review]

File: dump-objects (hunk 1 above)
Comment (TheBigLee) [reply dump-objects]: You are right. I rewrote this and added another parameter to the script. Afaik this makes more sense like this

File: dump-objects [hunk 2]
```diff
@@ -47,6 +49,7 @@ while getopts 'hvd:s' opt; do
     v) opt_verbose=yes ;;
     d) output_dir="$OPTARG" ;;
     s) opt_fastretries=yes ;;
+    D) opt_debug=true
```
Comment (mhutter) [inline dump-objects]:
```suggestion
    D) opt_debug=true ;;
```

**Review by mhutter — CHANGES_REQUESTED** [review]

File: dump-objects [hunk 3]
```diff
@@ -134,7 +139,7 @@ extract_versioned_apis() {
 }
 
 validate_kinds() {
-  local -a expected=( $(< /usr/local/share/k8s-object-dumper/must-exist) )
+  local -a expected=( "$(< /usr/local/share/k8s-object-dumper/must-exist)" )
```
Comment (mhutter) [inline dump-objects]: I know shellcheck complains about this, but is the behavior the same?

File: dump-objects [hunk 4]
```diff
@@ -208,7 +214,7 @@ object_kinds() {
 #
 #
 retrievable_kind() {
-  local -a known_to_fail=( $(< /usr/local/share/k8s-object-dumper/known-to-fail) )
+  local -a known_to_fail=( "$(< /usr/local/share/k8s-object-dumper/known-to-fail)" )
```
Comment (mhutter) [inline dump-objects]: Same as above

File: dump-objects (hunk 3 above)
Comment (TheBigLee) [reply dump-objects]: Ok it looks like this indeed doesn't work

Comment (TheBigLee) [reply dump-objects]: Fixed by using mapfile

**Review by mhutter — APPROVED** [review]

**Pull request merged**